package/MDAnalysis/coordinates/XYZ.py

```diff
@@ -303,7 +303,7 @@ def __init__(self, filename, **kwargs):
         # coordinates::core.py so the last file extension will tell us if it is
         # bzipped or not
         root, ext = os.path.splitext(self.filename)
-        self.xyzfile = util.anyopen(self.filename)
+        self._file = self.xyzfile = util.anyopen(self.filename)
```

Comment:
why do we keep this file handle around? This seems brittle? It also doesn't help our issue of too many open file handles.

Reply:
Readers have always worked this way - it's not that different from the XDRFile is it? Or does xdr open/close for each read?

Reply:
Yeah, we do. Except for when we use the iterator protocol. @dotsdl has implemented that years ago.  Besides, why have two file handles? I don't see the `self._file` in the base classes.